core/forms.py

```diff
@@ -45,7 +45,7 @@ def test_name_list():
     TEST_CHOICE = ()
 
     for test in tests:
-        data = ((test.test_name, test.test_name),)
+        data = ((Test.objects.get(id=test.id), test.test_name),)
```

Comment:
redundant Database request. When you already have all the test in "tests" so there is no need to get them again

Reply:
No sir actually I needed it as object., test.test_name was giving string which can't be referenced as foreign key as in model.I needed the object for of test. Related to this I did many changes, so actually it is correct I guess.

Reply:
test itself is an object if you need an object, use that instead of test.test_name

Reply:
yes I did this, It works.